middleware/supress_notfound.go

```diff
@@ -16,11 +16,20 @@ func SupressNotFound(router *chi.Mux) func(next http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			rctx := chi.RouteContext(r.Context())
+
+			// Match() updates rctx.
+			// Save the initial value to restore it later on for the next in chain.
+			initialRctx := *rctx
+
 			match := rctx.Routes.Match(rctx, r.Method, r.URL.Path)
 			if !match {
 				router.NotFoundHandler().ServeHTTP(w, r)
 				return
 			}
+
+			// Restore after modification.
+			*rctx = initialRctx
```

Comment:
The only thing that I'm unsure about is why does the `.Match()` method change the `rctx` in the first place.

This looks like a nice bugfix/workaround. However, can we look into the root cause in the `.Match()` method, please? I wouldn't expect it to change the route context at all.

Reply:
Never mind -- this is a documented behavior: https://github.com/go-chi/chi/blob/23c395f8524a30334126ca16fb4d37b88745b9b9/mux.go#L361-L362

Reply:
I think the right fix is to use `NewRouteContext()` per the above comment.

Reply:
@ezraisw thank you for the fix. Would you mind updating the PR to use `NewRouteContext()` instead of updating the top-level route context, please?